Lib/compileall.py

```diff
@@ -152,18 +152,17 @@ def compile_file(fullname, ddir=None, force=False, rx=None, quiet=0,
                           "in combination with stripdir or prependdir"))
 
     success = True
-    if quiet < 2 and isinstance(fullname, os.PathLike):
-        fullname = os.fspath(fullname)
+    fullname = os.fspath(fullname)
     name = os.path.basename(fullname)
 
     dfile = None
 
     if ddir is not None:
-        dfile = os.path.join(ddir, name)
+        dfile = os.path.join(os.fspath(ddir), name)
```

Comment:
One more nit: I think the call to `os.fspath` is only needed on line 165 with stripdir; os.path.join is happy to take PathLike objects

Reply:
Although actually neither works when `stripdir` is bytes, which the documentation says it's allowed to be, since `os.path.sep` is a  str

Reply:
I made the changes, let me know if there's anything else.